e2e-tests/portal/cypress/e2e/journeys/login/navigating-using-sign-in-link.spec.js

```diff
@@ -0,0 +1,178 @@
+    });
+
+    it('Opening a previously issued sign in link takes the user to the /login/sign-in-link-expired page to resend link and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [EXPIRED_SIGN_IN_TOKEN, NOT_EXPIRED_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: EXPIRED_SIGN_IN_TOKEN.signInTokenFromLink, userId: bank1Maker1Id });
+
+      linkShouldBeExpiredAndAbleToBeResent();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening a previously issued but not expired sign in link takes the user to the /login/sign-in-link-expired page to resend link and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [ANOTHER_NOT_EXPIRED_TOKEN, NOT_EXPIRED_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: ANOTHER_NOT_EXPIRED_TOKEN.signInTokenFromLink, userId: bank1Maker1Id });
+
+      linkShouldBeExpiredAndAbleToBeResent();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening the most recently issued, but expired, sign in link takes the user to the /login/sign-in-link-expired page to resend link and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [EXPIRED_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: EXPIRED_SIGN_IN_TOKEN.signInTokenFromLink, userId: bank1Maker1Id });
+
+      linkShouldBeExpiredAndAbleToBeResent();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening a sign in link with invalid token takes the user to the page not found page and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [INVALID_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: INVALID_SIGN_IN_TOKEN.signInTokenFromLink, userId: bank1Maker1Id }, { failOnStatusCode: false });
+      signInLink.shouldDisplayProblemWithServiceError();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+    it('Opening a sign in link with invalid username takes the user to the page not found page and does not give the user access to protected routes', () => {
+      cy.overridePortalUserSignInTokensByUsername({ username: BANK1_MAKER1.username, newSignInTokens: [NOT_EXPIRED_SIGN_IN_TOKEN] });
+      signInLink.visit({ token: NOT_EXPIRED_SIGN_IN_TOKEN.signInTokenFromLink, userId: 'notValidToken' }, { failOnStatusCode: false });
+      signInLink.shouldDisplayProblemWithServiceError();
+
+      checkUserDoesNotHaveAccessToProtectedRoutes();
+    });
+
+    it('Opening a sign in link with non-matching user id takes the user to the page not found page and does not give the user access to protected routes', () => {
```

Comment:
I've not got strong feelings here but slight preference for not found as if they have done something wrong without knowing it's good to know explicitly rather than just end up at the login page with no error back where they started

Reply:
This scenario refers to when the user id in query parameter is wrong, if I have understood this correctly we will not expect a legitimate user to be changing their user id in the magic link?

Reply:
As well as the scenario you described Abhi, this could occur when a user is signed into the session as one user but clicks on a sign in link for another